banhmi/src/controllers/auth.ts

```diff
@@ -0,0 +1,99 @@
+import { prisma } from "@/db";
+import { createAccessToken, getHashedPasswordWithPepper, getRandomString, refreshAccessToken, validatePassword } from "@/services/auth";
+import { Request, Response } from "express";
+
+const SALT_LENGTH = 50;
+
+async function registerHandler(_req: Request, res: Response) {
+    // TODO: middleware to check required fields and its type
+    const username = _req.body.username
+    // TODO: utils to return error codes
+    if (username == null) return res.status(400).send({
+        'error': 'username is required'
+    })
+    
+    const user = await prisma.user.findUnique({
+        where: {
+            username: username
+        }
+    })
+    if (user != null) return res.status(400).send({
+        'error': 'username already taken'
+    })
+
+    const salt = _req.body.salt
+    const hashedPassword = _req.body.hashedPassword
+    if (salt == null || hashedPassword == null) return res.status(400).send({
+        'error': 'salt and hashedPassword is required'
+    })
+
+    const hashedPasswordWithPepper = getHashedPasswordWithPepper(hashedPassword);
+    await prisma.user.create({
+        data: {
+            username: username,
+            salt: salt,
+            hashPassword: hashedPasswordWithPepper
+        }
+    })
+    res.send({
+        'username': username,
+        'salt': salt,
+        'hashedPassword': hashedPassword,
+        'pass': hashedPasswordWithPepper
+    })
+}
+
+// Get random salt for register
+async function getSaltHandler(_req: Request, res: Response) {
+    const salt = getRandomString(SALT_LENGTH);
+    res.send({'salt': salt});
+}
```

Comment:
As suggested by Minh, we will explicitly call `.status(200)` even when the return status code is 200. Besides, we will use `.json()` instead of `.send()`.

Reply:
Can you point out what is the different between `.json()` and `.send()` @minhnguyenphamquoc

Reply:
res.json({...}) eventually calls res.send(), but before that it:
- respects the json spaces and json replacer app settings
- ensures the response will have utf-8 charset and application/json Content-Type
@thienanhvu